### ISSUE TITLE
"Dynamic" softmax objective?

### ISSUE THREAD

**Issue by TristanTuscadero** [dmlc/xgboost#1854]
Currently, for multi-class classification, XGBoost supports a "fixed" label set, where separate trees are extracted per label. To handle "dynamic" label sets, one must do binary classification or a ranking classification. However it is often better to do softmax classification over a label set.

For example, consider a task where I give you a natural language text.

Passage: "Sally gave the ball to John. John threw the ball to Timmy. Timmy took the ball home."
Question: "Who has the ball?"
Candidates = [Sally, John, Timmy]
Label = Timmy

We would extract a feature vector for each candidate:

Sally = ['ball'-in-question-and-ball-after-name, is-first-name-mentioned, ...] 
John = ['ball'-in-question-and-'ball'-after-name, is-second-name-mentioned, ...]
Timmy = ['ball'-in-question-and-'ball'-before-name, last-instance-with-'ball'...,]

Right now the best we could do is just do a logistic regression 3 different training samples and Timmy = 1, Sally = 0, John = 0 _or_ we could do a ranking objective where Timmy = 0, Sally = 1, John = 1. But for cases like this it is often better to do a softmax over the candidates (especially when the number of candidates is large.), since we know there is exactly 1 correct answer and we want a probability distribution.

This should be fairly straightforward to do with the "group" mechanism by following the existing code for ranking objective + softmax, correct? I can add it myself in my fork if this is the case, but I want to make sure I'm not missing something.

**Comment by pommedeterresautee** [comment]
Old proposition, no answer. Closing
